cmake/winml.cmake

```diff
@@ -442,15 +411,14 @@ set_target_properties(winml_dll
 set_target_properties(winml_dll
   PROPERTIES
   LINK_FLAGS
-  "/DEF:${WINML_DIR}/windows.ai.machinelearning.def ${os_component_link_flags} /DELAYLOAD:d3d12.dll /DELAYLOAD:d3d11.dll /DELAYLOAD:dxgi.dll")
+  "/DEF:${WINML_DIR}/windows.ai.machinelearning.def ${os_component_link_flags} /DELAYLOAD:d3d12.dll /DELAYLOAD:d3d11.dll /DELAYLOAD:dxgi.dll /DELAYLOAD:directml.dll")
```

Comment:
winml_dll shouldn't depend on directml.dll should it? why do we need the delay load?

Reply:
IsFloat16Supported calls into DML in order to learn if float16 is supported or not to decide what it should do around session creation (fail or not if fp16 was asked for).

---
In reply to: [354034322](https://github.com/microsoft/onnxruntime/pull/2534#discussion_r354034322) [](ancestors = 354034322)

Reply:
i added it as a //todo on your whiteboard list.

---
In reply to: [354589957](https://github.com/microsoft/onnxruntime/pull/2534#discussion_r354589957) [](ancestors = 354589957,354034322)